jetty-webapp/src/test/java/org/eclipse/jetty/webapp/WebAppContextTest.java

```diff
@@ -254,177 +266,196 @@ public void testNullPath() throws Exception
         server.addConnector(connector);
 
         server.start();
-        try
-        {
-            String response = connector.getResponse("GET http://localhost:8080 HTTP/1.1\r\nHost: localhost:8080\r\nConnection: close\r\n\r\n");
-            assertThat("Response OK", response, containsString("200 OK"));
-        }
-        finally
-        {
-            server.stop();
-        }
+
+        String rawResponse = connector.getResponse("GET http://localhost:8080 HTTP/1.1\r\nHost: localhost:8080\r\nConnection: close\r\n\r\n");
+        HttpTester.Response response = HttpTester.parseResponse(rawResponse);
+        assertThat("Response OK", response.getStatus(), is(HttpStatus.OK_200));
     }
 
     @Test
     public void testNullSessionAndSecurityHandler() throws Exception
     {
-        Server server = new Server(0);
+        Server server = newServer();
+
         HandlerList handlers = new HandlerList();
         ContextHandlerCollection contexts = new ContextHandlerCollection();
         WebAppContext context = new WebAppContext(null, null, null, null, null, new ErrorPageErrorHandler(),
             ServletContextHandler.NO_SESSIONS | ServletContextHandler.NO_SECURITY);
         context.setContextPath("/");
-        context.setBaseResource(Resource.newResource("./src/test/webapp"));
+
+        Path testWebapp = MavenTestingUtils.getProjectDirPath("src/test/webapp");
+        context.setBaseResource(new PathResource(testWebapp));
         server.setHandler(handlers);
         handlers.addHandler(contexts);
         contexts.addHandler(context);
 
         LocalConnector connector = new LocalConnector(server);
         server.addConnector(connector);
 
-        try
-        {
-            server.start();
-            assertTrue(context.isAvailable());
-        }
-        finally
-        {
-            server.stop();
-        }
+        server.start();
+        assertTrue(context.isAvailable());
     }
 
-    class ServletA extends GenericServlet
+    @Test
+    public void ordering() throws Exception
     {
-        @Override
-        public void service(ServletRequest req, ServletResponse res) throws ServletException, IOException
-        {
-            this.getServletContext().getContext("/A/s");
-        }
+        WebAppContext context = new WebAppContext();
+        Path testWebappDir = MavenTestingUtils.getProjectDirPath("src/test/webapp");
+        context.setBaseResource(new PathResource(testWebappDir));
+        context.setContextPath("/test");
+        new WebInfConfiguration().preConfigure(context);
+        assertEquals(Arrays.asList("acme.jar", "alpha.jar", "omega.jar"),
```

Comment:
I find assertThat with a contains matcher more comprehensible for list checks

Reply:
That's an old test, I'll take another pass through for cleanup.

From jetty-10.0.x HEAD ...

https://github.com/eclipse/jetty.project/blob/ccdd20b7bc8fdf2b54727f3077837bc6de043fde/jetty-webapp/src/test/java/org/eclipse/jetty/webapp/WebAppContextTest.java#L485-L486

Reply:
This ordering() test case attempts to prove ClassLoader jar file order.
It's been removed.